src/environment_setup/biopam.ml

```diff
@@ -4,172 +4,250 @@ Provide tools via Biopam: https://github.com/solvuu/biopam
-  let root ~install_path = dir ~install_path // ".opam"
+  let root ~install_path name = dir ~install_path // "opam-root-" ^ name
 
   (* TODO:
      Instead of just making sure that this file exists? Wouldn't it be better
      to make sure that a command from this program gives the right output?
      ie. $ opam --version = 1.2.2 *)
-  let target ?host ~install_path =
-    K.single_file ?host (bin ~install_path)
+  let target ~host ~install_path =
+    KEDSL.single_file ~host (bin ~install_path)
 
   (* A workflow to ensure that opam is installed. *)
-  let installed ?host ~install_path =
+  let installed ~(run_program : Machine.Make_fun.t) ~host ~install_path =
     let url = default_opam_url in
-    let opam_exec   = target ?host ~install_path in
+    let opam_exec   = target ~host ~install_path in
     let install_dir = dir ~install_path in
-    K.workflow_node opam_exec
+    let open KEDSL in
+    workflow_node opam_exec
       ~name:"Install opam"
-      ~make:(K.daemonize ?host
-        K.Program.(
-          exec ["mkdir"; "-p"; install_dir]
-          && exec ["cd"; install_dir]
-          && Workflow_utilities.Download.wget_program ~output_filename:"opam" url
-          && shf "chmod +x %s" opam_exec#path))
-      ~edges:[K.on_failure_activate
-                (Workflow_utilities.Remove.path_on_host
-                   ~host:K.Host.tmp_on_localhost install_dir)]
-
-  let kcom ?(switch=true) ~install_path k fmt =
+      ~make:(
+        run_program
+          ~requirements:[
+            `Internet_access;
+            `Self_identification ["opam"; "ini"];
+          ]
+          Program.(
+            exec ["mkdir"; "-p"; install_dir]
+            && exec ["cd"; install_dir]
+            && Workflow_utilities.Download.wget_program ~output_filename:"opam" url
+            && shf "chmod +x %s" opam_exec#path))
+      ~edges:[
+        on_failure_activate
+          (Workflow_utilities.Remove.path_on_host ~host install_dir);
+      ]
+
+  let kcom ~root_name ~install_path k fmt =
     let bin = bin ~install_path in
-    let root = root ~install_path in
-    (* Pass the ROOT and SWITCH args as environments to not disrupt the flow of
-       the rest of the arguments:
-        ie opam --root [root] init -n
+    let root = root ~install_path root_name in
+    (* Pass the ROOT args as environments to not disrupt the flow of the rest
+       of the arguments: ie `opam --root [root] init -n`
        doesn't parse correctly. *)
-    if switch then
-      ksprintf k ("OPAMROOT=%s OPAMSWITCH=0.0.0 %s " ^^ fmt) root bin
-    else
-      ksprintf k ("OPAMROOT=%s %s " ^^ fmt) root bin
-
-  let program_sh ?switch ~install_path fmt =
-    kcom ?switch ~install_path K.Program.sh fmt
-
-  let command_shell ?switch ?host ~install_path fmt =
-    kcom ?switch ~install_path (K.Command.shell ?host) fmt
+    ksprintf k
+      ("PATH=%s:$PATH OCAMLRUNPARAM=b OPAMLOCKRETRIES=20000 OPAMBASEPACKAGES= OPAMYES=true \
```

Comment:
Could you comment what these env vars mean; I know we can just look them up, but it might be useful in the future?